jsonschema-core.xml

```diff
@@ -1394,87 +1412,179 @@
-                            <!-- All of the standard meta-schemas use an empty fragment in their id/$id values. -->
+                            The root schema of a JSON Schema document SHOULD contain an "$id" keyword
+                            with an <xref target="RFC3986">absolute-URI</xref> (containing a scheme,
+                            but no fragment).
                         </t>
                     </section>
-                    <section title="Changing the base URI within a schema file">
+                    <section title="JSON Pointer fragments and embedded schema resources"
+                             anchor="embedded">
                         <t>
-                            When an "$id" sets the base URI, the object containing that "$id" and all of
-                            its subschemas can be identified by using a JSON Pointer fragment starting
-                            from that location.  This is true even of subschemas that further change the
-                            base URI.  Therefore, a single subschema may be accessible by multiple URIs,
-                            each consisting of base URI declared in the subschema or a parent, along with
-                            a JSON Pointer fragment identifying the path from the schema object that
-                            declares the base to the subschema being identified.  Examples of this are
-                            shown in section <xref target="idExamples" format="counter"></xref>.
+                            Since JSON Pointer URI fragments are constructed based on the structure
+                            of the schema document, an embedded schema resource and its subschemas
+                            can be identified by JSON Pointer fragments relative to either its own
+                            canonical URI, or relative to the containing resource's URI.
                         </t>
-                    </section>
-                    <section title="Location-independent identifiers">
                         <t>
-                            Using JSON Pointer fragments requires knowledge of the structure of the schema.
-                            When writing schema documents with the intention to provide re-usable
-                            schemas, it may be preferable to use a plain name fragment that is not tied to
-                            any particular structural location.  This allows a subschema to be relocated
-                            without requiring JSON Pointer references to be updated.
+                            Conceptually, a set of linked schemas should behave identically whether
+                            each schema is a separate document connected with
+                            <xref target="references">schema references</xref>, or is structured as
+                            a single document with one or more schema resources embedded as
+                            subschemas.
```

Comment:
Is it worth mentioning "as per section (https://github.com/json-schema-org/json-schema-spec/blob/9ac32f405ccadc48fa0ed732839f29d49832aec1/jsonschema-core.xml#L1458-L1465)" (whatever section that is)?

I say this, because when reading this section, I'm left wondering when I (as a schema author) should use `$id` vs `$anchor`

Reply:
@Relequestual I don't understand what this paragraph has to do with `$anchor` at all?

Keep in mind that when it comes to sorting out people's confusion migrating from the old `$id`, that goes on the web site.  The spec is primarily to be read on its own.  I can't see any reason why someone reading this for the first time would think it has anything to do with `$anchor`.

Reply:
`$anchor` has only previously been mentioned once, in the section on fragments, with a forward reference.  It's not properly introduced until several sections after this.

Reply:
Fair enough.